src/Symfony/Component/DependencyInjection/Dumper/XmlDumper.php

```diff
@@ -265,6 +265,11 @@ private function convertParameters(array $parameters, string $type, \DOMElement
     {
         $withKeys = !array_is_list($parameters);
         foreach ($parameters as $key => $value) {
+            if (str_starts_with($key, '.')) {
```

Comment:
I'm not sure about those checks in dumpers: to me, the build-time parameters should be already removed before dumpers are called. This means we should go with an after-removing compiler pass to remove parameters.

Reply:
It seems to be working with a compiler pass 🎉, but one test is failing though and I cannot understand why, I'll try to look again later.

Reply:
Fixed!